packetizer.go

```diff
@@ -1,64 +1,89 @@
 package rpc
 
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	"github.com/ugorji/go/codec"
+)
+
 type packetizer interface {
-	Packetize() error
+	NextFrame() (RPCMessage, error)
 }
 
 type packetHandler struct {
-	receiver receiver
-	dec      byteReadingDecoder
+	dec           decoder
+	reader        io.Reader
+	frameDecoders decoderPool
+	protocols     *protocolHandler
+	calls         *callContainer
+	mtx           sync.Mutex
```

Comment:
what exactly is this mutex protecting? I only see it used in loadNextFrame, which AFAICT is used only synchronously